### PR TITLE
Add resize doc for both filesystem and block mode PVCs

### PR DESCRIPTION
**Pull request by humblec**
Signed-off-by: Humble Chirammal <hchiramm@redhat.com>

### CODE REVIEW

**Comment by humblec** [comment]
@Madhu-1 @ShyamsundarR @nixpanic ptal

File: docs/resize-rbd.md [hunk 1]
```diff
@@ -0,0 +1,277 @@
+# Resize RBD PVCs
+
```
Comment (Madhu-1) [inline docs/resize-rbd.md]: Add required feature gates need to be enabled to support CSI resize

File: docs/resize-rbd.md [hunk 2]
```diff
@@ -0,0 +1,277 @@
...
+- [Resize RBD PVCs](#resize-rbd-pvcs)
```
Comment (Madhu-1) [inline docs/resize-rbd.md]: Add Minimum kubernetes version required to resize filesystem and block  PVC

File: docs/resize-rbd.md [hunk 3]
```diff
@@ -0,0 +1,277 @@
...
+    capacity:
+      storage: 1Gi
+    phase: Bound
+kind: List
+metadata:
+  resourceVersion: ""
+  selfLink: ""
+```
+
+Directory size in  pod using this PVC
```
Comment (Madhu-1) [inline docs/resize-rbd.md]: Add this as bullet point?

File: docs/resize-rbd.md [hunk 4]
```diff
@@ -0,0 +1,277 @@
...
+
+- [Resize RBD PVCs](#resize-rbd-pvcs)
+  - [Filesystem resize on RBD filesystem volume mode PVCs](#filesystem-resize-on-rbd-filesystem-volume-mode-pvcs)
+  - [Resize Block PVC](#resize-block-pvc)
+
+## Filesystem resize on RBD filesystem volume mode PVCs
+
+pvc.yaml
+
+```
```
Comment (Madhu-1) [inline docs/resize-rbd.md]: console

File: docs/resize-rbd.md [hunk 5]
```diff
@@ -0,0 +1,277 @@
...
+/dev/rbd0       976M  2.6M  958M   1% /var/lib/www/html
+tmpfs           1.9G   12K  1.9G   1% /run/secrets/kubernetes.io/serviceaccount
+tmpfs           1.9G     0  1.9G   0% /proc/acpi
+tmpfs           1.9G     0  1.9G   0% /proc/scsi
+tmpfs           1.9G     0  1.9G   0% /sys/firmware
+#
+
+```
+
+Now resize the PVC by editing the PVC (pvc.spec.resource.requests.storage)
```
Comment (Madhu-1) [inline docs/resize-rbd.md]: Add this as bullet point?

File: docs/resize-rbd.md [hunk 6]
```diff
@@ -0,0 +1,277 @@
...
+      type: FileSystemResizePending
+    phase: Bound
+kind: List
+metadata:
+  resourceVersion: ""
+  selfLink: ""
+
+```
+
+Now you  can see the pvc stats as FileSystemResizePending, once the kubelet
```
Comment (Madhu-1) [inline docs/resize-rbd.md]: `pvc stats` to `pvc status`

File: docs/resize-rbd.md [hunk 7]
```diff
@@ -0,0 +1,277 @@
...
+calls the NodeExpandVolume to resize the PVC  on node, the status conditions
```
Comment (Madhu-1) [inline docs/resize-rbd.md]: change `the status conditions  will be updated and status will be updated` **to** `the status conditions
will be updated and `status.capacity.storage` will be updated`

File: docs/resize-rbd.md [hunk 8]
```diff
@@ -0,0 +1,277 @@
...
+will be updated and status will be updated
+
+```bash
+[$]kubectl get pvc
+NAME      STATUS   VOLUME                                     CAPACITY   ACCESS MODES   STORAGECLASS      AGE
+rbd-pvc   Bound    pvc-efe688d6-a420-4041-900e-c5e19fd73ebf   10Gi       RWO            csi-rbd-sc   7m6s
+
+```
+
+Now let us check the Directory  size inside the pod where PVC is mounted
```
Comment (Madhu-1) [inline docs/resize-rbd.md]: bullet point?

Comment (Madhu-1) [reply docs/resize-rbd.md]: double space between Directory and size

File: docs/resize-rbd.md [hunk 9]
```diff
@@ -0,0 +1,277 @@
...
+shm              64M     0   64M   0% /dev/shm
+/dev/rbd0       9.9G  4.5M  9.8G   1% /var/lib/www/html
+tmpfs           1.9G   12K  1.9G   1% /run/secrets/kubernetes.io/serviceaccount
+tmpfs           1.9G     0  1.9G   0% /proc/acpi
+tmpfs           1.9G     0  1.9G   0% /proc/scsi
+tmpfs           1.9G     0  1.9G   0% /sys/firmware
+#
+```
+
+now you  can see the size of `/var/lib/www/html` is updated from 976M to 9.9G
```
Comment (Madhu-1) [inline docs/resize-rbd.md]: double space between you and can

File: docs/resize-rbd.md [hunk 10]
```diff
@@ -0,0 +1,277 @@
...
+  volumeName: pvc-54885275-7ca9-4b89-8e7e-c99f375d1174
+status:
+  accessModes:
+  - ReadWriteOnce
+  capacity:
+    storage: 1Gi
+  phase: Bound
+```
+
+Device size in pod using this PVC
```
Comment (Madhu-1) [inline docs/resize-rbd.md]: bullet point

File: docs/resize-rbd.md [hunk 11]
```diff
@@ -0,0 +1,277 @@
...
+```bash
+[$]kubectl exec -it pod-with-raw-block-volume sh
+sh-4.4# blockdev --getsize64 /dev/xvda
+1073741824
+```
+
+rbd Block PVC is mounted at `/dev/xvda` of the pod, and the size is `1073741824`
+bytes which is equal to `1Gib`
+
+Now resize the PVC by editing the PVC (pvc.spec.resource.requests.storage)
```
Comment (Madhu-1) [inline docs/resize-rbd.md]: bullet point?

Comment (Madhu-1) [reply docs/resize-rbd.md]: Now resize the PVC by editing the PVC (pvc.spec.resource.requests.storage) to the new size which should be greater than the current size?

File: docs/resize-rbd.md [hunk 12]
```diff
@@ -0,0 +1,277 @@
...
+  resourceVersion: "7923"
+  selfLink: /api/v1/namespaces/default/persistentvolumeclaims/raw-block-pvc
+  uid: 54885275-7ca9-4b89-8e7e-c99f375d1174
+spec:
+  accessModes:
+  - ReadWriteOnce
+  resources:
+    requests:
+      storage: 10Gi
+  storageClassName: rook-ceph-block
```
Comment (Madhu-1) [inline docs/resize-rbd.md]: change storageclassname from rook-ceph-block to csi-rbd-sc

File: docs/resize-rbd.md [hunk 13]
```diff
@@ -0,0 +1,277 @@
...
+    message: Waiting for user to (re-)start a pod to finish file system resize of
+      volume on node.
+    status: "True"
+    type: FileSystemResizePending
+  phase: Bound
+
+```
+
+Now you can see the pvc stats as FileSystemResizePending, once the kubelet calls
+the NodeExpandVolume to resize the PVC on node, the status conditions will be updated
```
Comment (Madhu-1) [inline docs/resize-rbd.md]: change the status conditions will be updated and status will be updated **to** the status conditions will be updated and status.capacity.storage will be updated

File: docs/resize-rbd.md (hunk 2 above)
Comment (nixpanic) [reply docs/resize-rbd.md]: > kubernetes version running in your cluster should be >= v1.16

it is there (now)

File: docs/resize-rbd.md [hunk 14]
```diff
@@ -0,0 +1,283 @@
...
+metadata:
+  resourceVersion: ""
+  selfLink: ""
+```
+
+- Directory size in pod using this PVC
+
+```
+kubectl exec -it csi-rbd-demo-pod sh
+# df -h
```
Comment (nixpanic) [inline docs/resize-rbd.md]: You only need to show the `/var/lib/www/html` directory, so run `df -h /var/lib/www/html` to skip the unneeded (and possibly confusing) output.

File: docs/resize-rbd.md [hunk 15]
```diff
@@ -0,0 +1,283 @@
...
+NAME      STATUS   VOLUME                                     CAPACITY   ACCESS MODES   STORAGECLASS      AGE
+rbd-pvc   Bound    pvc-efe688d6-a420-4041-900e-c5e19fd73ebf   10Gi       RWO            csi-rbd-sc   7m6s
+
+```
+
+- Now let us check the directory size inside the pod where PVC is mounted
+
+```bash
+ kubectl exec -it csi-rbd-demo-pod sh
+# df -h
```
Comment (nixpanic) [inline docs/resize-rbd.md]: `df -h /var/lib/www/html` here too please

File: docs/resize-rbd.md [hunk 16]
```diff
@@ -0,0 +1,283 @@
...
+tmpfs           1.9G     0  1.9G   0% /proc/scsi
+tmpfs           1.9G     0  1.9G   0% /sys/firmware
+#
+
+```
+
+- Now resize the PVC by editing the PVC (pvc.spec.resource.requests.storage)
+
+```bash
+#kubectl edit pvc rbd-pvc
```
Comment (nixpanic) [inline docs/resize-rbd.md]: this does not need to run as root, you have the `[$]` prefix for almost all other `kubectl` commands.

File: docs/resize-rbd.md [hunk 17]
```diff
@@ -0,0 +1,283 @@
...
+[$]kubectl get pvc
+NAME      STATUS   VOLUME                                     CAPACITY   ACCESS MODES   STORAGECLASS      AGE
+rbd-pvc   Bound    pvc-efe688d6-a420-4041-900e-c5e19fd73ebf   10Gi       RWO            csi-rbd-sc   7m6s
+
+```
+
+- Now let us check the directory size inside the pod where PVC is mounted
+
+```bash
+ kubectl exec -it csi-rbd-demo-pod sh
```
Comment (nixpanic) [inline docs/resize-rbd.md]: add the `[$]` prefix too please.

File: docs/resize-rbd.md [hunk 18]
```diff
@@ -0,0 +1,283 @@
...
+    capacity:
+      storage: 1Gi
+    phase: Bound
+kind: List
+metadata:
+  resourceVersion: ""
+  selfLink: ""
+```
+
+- Directory size in pod using this PVC
```
Comment (nixpanic) [inline docs/resize-rbd.md]: "Directory" -> "mounted Filesystem" maybe?

File: docs/resize-rbd.md (hunk 17 above)
Comment (humblec) [reply docs/resize-rbd.md]: Done,

File: docs/resize-rbd.md (hunk 18 above)
Comment (humblec) [reply docs/resize-rbd.md]: Done.

File: docs/resize-rbd.md (hunk 16 above)
Comment (humblec) [reply docs/resize-rbd.md]: Done.

File: docs/resize-rbd.md (hunk 15 above)
Comment (humblec) [reply docs/resize-rbd.md]: Done.

File: docs/resize-rbd.md (hunk 14 above)
Comment (humblec) [reply docs/resize-rbd.md]: Done.

**Comment by humblec** [comment]
@Madhu-1 @nixpanic comments are addressed. PTAL

File: docs/resize-rbd.md [hunk 19]
```diff
@@ -0,0 +1,265 @@
...
+    lastTransitionTime: "2019-12-19T06:02:15Z"
+    message: Waiting for user to (re-)start a pod to finish file system resize of
+      volume on node.
+    status: "True"
+    type: FileSystemResizePending
+  phase: Bound
+
+```
+
+Now you can see the pvc stats as FileSystemResizePending, once the kubelet calls
```
Comment (Madhu-1) [inline docs/resize-rbd.md]: need o change  `pvc stats` to `pvc status`

**Comment by Madhu-1** [comment]
> The filename can be resize-pvc.md we can add a section for both cephfs and rbd in the same file

@humblec the above comment need to be handled

**Comment by humblec** [comment]
@Madhu-1 last set of comments are addressed. PTAL

**Pull request merged**